nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -734,32 +728,30 @@ def _aggregation_fn(
         return self._tensor_processor.masked_median(stacked_samples, axis=axis, mask=mask, keepdims=keepdims)
 
 
-class MedianAbsoluteDeviationAggregator(TensorAggregatorBase):
+class MedianAbsoluteDeviationAggregator(AggregatorBase):
     def _register_reduced_input_impl(self, x: TensorType) -> None:
         return self._container.append(x)
 
     def _aggregate_impl(self) -> Dict[str, NNCFTensor]:
         stacked_val = self._tensor_processor.stack(self._container)
-
         mask = self._tensor_processor.zero_elements(stacked_val)
         median_per_ch = self._tensor_processor.masked_median(
             stacked_val, mask=mask, axis=self._aggregation_axes, keepdims=True
         )
-
         mad_values = self._tensor_processor.median(
             self._tensor_processor.abs(self._tensor_processor.sub(stacked_val, median_per_ch)),
             axis=self._aggregation_axes,
             keepdims=self._keepdims,
         )
-        if not self._keepdims:
-            median_per_ch = self._tensor_processor.squeeze(median_per_ch, self._aggregation_axes)
+        squeezed_mad_values = self._tensor_processor.squeeze(mad_values, 0)
+        squeezed_median_per_ch = self._tensor_processor.squeeze(median_per_ch, 0)
```

Comment:
Please leave branch with `not self._keepdims`, I'm planning to expand testing with different `self._keepdims` values, so next time we will need to change it we shouldn't change any test

Reply:
Rebased, done